### ISSUE TITLE
[New build - DEFAULT] 24w27a: Sadde and pedestal, mech improvements

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5848]
This build features several improvements for mech following tester feedback.

A new saddle and pedestal seat have been added.

Mech improvements and fixes:
- Can now freelook
- #5845 3rd person view improved
- Symmetry logic has been changed, so now the animation no longer requires a key at 0.5
- new "Set idle position from current angles" button
- New "Is Symmetry Inverted" param in rotator

**Comment by Kaiser-Indrasil** [comment]
I'm not sure what's the point of the Saddle to be honest - we already have regular chairs which provide the exact same seating pose.

![image](https://github.com/tsunamayo/Starship-EVO/assets/63756057/bf937c77-72ca-468e-9c49-51c0fad79b75)

Sure, it doesn't stick out from the back like the seats do, but the main application point would be bikes.
But on a bike, you sit at a 45 degree angle forward with your legs spread apart, like this:

![image](https://github.com/tsunamayo/Starship-EVO/assets/63756057/60b4489f-9628-4e4e-abb8-c064e9049a84)

Sure, we could already use Ottomans on a hinge to make a similar effect, but then the player's camera and the character's face are facing downwards:

![image](https://github.com/tsunamayo/Starship-EVO/assets/63756057/319b1ba7-a1a2-4e00-9c2e-026b807a2016)

Now, Standing Seat makes sense because it is small AND provides a new "seating" pose
Saddle could use one too

![image](https://github.com/tsunamayo/Starship-EVO/assets/63756057/2b341360-a16e-48ec-9dee-91b67b1a4bf3)

**Comment by tsunamayo** [comment]
Well I am waiting for more people to chime in, if people agree I could try to tweak the pose. For now it is an easy rider one...

**Comment by Tackietacktack** [comment]
Yeah, 45°degree are good for a racing bike but by Harley, horses or so  its too much.
Spreaded legs were pretty so that we get a 25cm brick between.

**Comment by ProPeach** [comment]
I think it would  be cool to have a slight angle forward.
Interestingly if you setup a yoke, the idle animatoin actually leans the character forward a little bit temporarily which does look nice. 

https://github.com/tsunamayo/Starship-EVO/assets/27857711/d1ef57b6-e93f-4f73-ad45-2e9cad9478c0

![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/fee7dab2-f084-4050-bfd0-d71e153e104e)

Would it be posible to link the amount of bend at the characters waist  to how far away the yoke is? That would mean you could have an upright pose or an angled pose for the same seat



One thing that I do think would improve all the poses on the seats is moving the character back by ~15cm, so that they actually sit with their back against the seat. There is a lot of dead/empty space with the current poses.
![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/01c8644f-3aa0-40c3-a844-5a1b63c06b73)


![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/f6d7ae5d-7732-47b9-beb9-8e01fed86e00)

**Comment by Crimson-Artist** [comment]
I think the current saddle seat should have more open legs like you are actually straddling it. I also think there should also be a another version of the saddle seat that is turned 90 degrees and keeps the current sitting animation.

Also I found a bug. Standing seats that are linked to a yoke will cause the character model to teleport and clip into the parent entity as opposed to teleporting to the standing spot.

**Comment by Tackietacktack** [comment]
I don't know if it is a alternate but if it is possible to combinate seat and pedestal seat, which make possible to change the leg degree.
Like pedals.

**Comment by Crimson-Artist** [comment]
I think there should be an easier way to enter standing seats. Right now they are so small and subtle that people might not notice it as the thing we have to interact with. So I was thinking that adding some kind of logic component where if you connect a logic input like a button a standing seat pressing the button while standing about 1/2m from the seat will teleport the player into the seat. This will make it easier for certain builds like tight exo suits that only open from the top.

**Comment by ZachZent** [comment]
Actual saddle has you straddling the seat, not sitting in it.
![image](https://github.com/tsunamayo/Starship-EVO/assets/36980446/8f1f34ff-0124-4ba6-b283-b4fbe5b576fb)

**Comment by BigBadKangaroo** [comment]
Hi there,
I have to agree with the seating animation of the new saddle is a bit weird and doesn't really fit it's style / use purpose.

I made two sketches of what I would expect from the saddle: 
![Chopper_pos_back](https://github.com/tsunamayo/Starship-EVO/assets/81495956/335d7fff-3294-4419-ad3e-6f53a98ef151)
![Chopper_pos_side](https://github.com/tsunamayo/Starship-EVO/assets/81495956/3fc9a574-ae27-4ba3-9eb5-761f9644d758)

In these skecthes the player model sits a bit further back in the seat, leans forward and has angled legs to create a more realistic pose. The legs spread so a 1/4th brick fits easily between the legs (maybe even a bit more?). The arms grab the handles and reach forward to do so.
(Small thing to notice; the player grabs the handles "inverted". Normaly while riding a bike for example, the fingers point downwards and not upwards like they're now. Possible chance for a "bike style" yoke? 

Something that bothers me is that the character doesn't lean forward to grab the handles if it doesn't reach while sitting normally. I had to place the handles 1/8th brick closer to even reach it:
![20240702195726_1](https://github.com/tsunamayo/Starship-EVO/assets/81495956/d7a12f56-1cce-4723-9e1d-e20857dd8471)

Here's the BP in case you want to get a look at it. It's just a basic concept:
[Chopper.zip](https://github.com/user-attachments/files/16072745/Chopper.zip)


I tested the standing seat aswell, works great and gives me ideas for locking players in position for open spaces on hovercrafts or spaceships, i.e. troop carriers where the troops stand in the loading bay (Republic gunship from Star Wars etc) 
I can imagine that some players might find it difficult to recognize this seat, if they didn't build the ship themselves and didn't know where to find it. However, I don't really have an idea on how to "improve" that without breaking it's usefulnes. It's a great seat and does what I expected from it.

In general I really like these 2 new seats. One of them just needs a more polished seating animation :)

**Comment by tsunamayo** [comment]
@ZachZent Sorry I forgot to update you, I had repushed the build with the fix for standing.

**Comment by tsunamayo** [comment]
In next update:

![image](https://github.com/tsunamayo/Starship-EVO/assets/6604318/a6480d0b-fee4-42ca-943e-1dfa12f22b24)

**Comment by tsunamayo** [comment]
@ProPeach problem is that updating this would break all current ship design...

**Comment by ZachZent** [comment]
@tsunamayo Can still make a new chair and keep the old chair, but hide it in the codex

**Comment by ProPeach** [comment]
@tsunamayo would it just move the player back in existing seats, or actually break things so that those seats no longer work?

If it just moves the player position in old builds, I would be absolutely fine with that! There have been a few issues raised on Github over the years about moving the seat position back so I think you'll have popular support if you do

**Comment by SovietSponge** [comment]
Personally would like seeing seats changed, at worst the characters would sit a bit far back in my fake seat cushions and it'd be far easier to build compact tiny builds if you didn't have to account for about 25cm of space extra simply due to the odd seating positions of current seats.
